File: linked-lists/2.c
/**
 * 2.c
 * Linked Lists
 * https://github.com/ashenm/dsa
 *
 * Demonstrates linked list itteration
 *
 * Ashen Gunaratne
 * mail@ashenm.ml
 *
 */

#include <stdio.h>
#include <stdlib.h>

typedef struct node
{
    int value;
    struct node *next;
} node;

int main (void)
{

    node *x = malloc(sizeof(node));
    node *y = malloc(sizeof(node));

    if (x == NULL || y == NULL)
    {
        return 1;
    }

    x -> value = 25;
    x -> next = y;

    y -> value = 50;
    y -> next = NULL;

    for (node *n = x; n != NULL; n = n -> next)
    {
        printf("%i\n", n -> value);
    }

    free(x);
    free(y);

}

// vim: set expandtab shiftwidth=4:

File: linked-lists/2.js
/**
 * 2.js
 * Linked Lists
 * https://github.com/ashenm/dsa
 *
 * Demonstrates linked list itteration
 *
 * Ashen Gunaratne
 * mail@ashenm.ml
 *
 */

'use strict';

class Node {

  constructor (value, next=null) {
    this.value = value;
    this.next = next;
  }

}

const main = function () {

  const y = new Node(50);
  const x = new Node(25, y);

  for (let n = x; n !== null; n = n.next) {
    console.log(n.value);
  }

};

if (require.main === module) {
  main();
}

// vim: set expandtab shiftwidth=2:
